manage_breast_screening/auth/services.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from typing import Any, Dict
+
+from authlib.jose import JoseError, jwt
+from authlib.jose.errors import (
+    ExpiredTokenError,
+    InvalidClaimError,
+    InvalidTokenError,
+    MissingClaimError,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class InvalidLogoutToken(Exception):
+    """Raised when a CIS2 back-channel logout token is invalid."""
+
+    def __init__(self, cause=None):
+        """
+        Initialize with optional cause.
+
+        Args:
+            cause: The original exception that caused this error
+        """
+        self.cause = cause
+        super().__init__(str(cause) if cause else "Invalid logout token")
+
+
+class DecodeLogoutToken:
+    def call(
```

Comment:
I think I would just make this a regular function if there are no other methods and no instance members.

Otherwise, you can define `__call__` to make an object callable directly, i.e. `obj(...)` rather than `obj.call`